### PR TITLE
MINOR: log broker configs in KRaft mode

### PR DESCRIPTION
**Pull request by cmccabe**

### CODE REVIEW

**Review by niket-goel — APPROVED** [review]

**Comment by cmccabe** [comment]
In the ZK-based broker code in `KafkaServer.scala`, the configs are logged from here:

```
    at org.apache.kafka.common.config.AbstractConfig.logAll(AbstractConfig.java:370)
    at org.apache.kafka.common.config.AbstractConfig.<init>(AbstractConfig.java:121)
    at org.apache.kafka.common.config.AbstractConfig.<init>(AbstractConfig.java:146)
    at kafka.server.KafkaConfig.<init>(KafkaConfig.scala:1381)
    at kafka.server.DynamicBrokerConfig.processReconfiguration(DynamicBrokerConfig.scala:531)
    at kafka.server.DynamicBrokerConfig.updateCurrentConfig(DynamicBrokerConfig.scala:520)
    at kafka.server.DynamicBrokerConfig.$anonfun$updateDefaultConfig$1(DynamicBrokerConfig.scala:307)
    at kafka.server.DynamicBrokerConfig.updateDefaultConfig(DynamicBrokerConfig.scala:303)
    at kafka.server.DynamicBrokerConfig.initialize(DynamicBrokerConfig.scala:213)
    at kafka.server.KafkaServer.startup(KafkaServer.scala:228)
    at kafka.Kafka$.main(Kafka.scala:109)
    at kafka.Kafka.main(Kafka.scala)
```

File: core/src/main/scala/kafka/server/BrokerServer.scala [hunk 1]
```diff
@@ -389,6 +389,9 @@ class BrokerServer(
       // a potentially lengthy recovery-from-unclean-shutdown operation here, if required.
       metadataListener.startPublishing(metadataPublisher).get()
 
+      // Log static broker configurations.
+      new KafkaConfig(config.originals(), true)
```
Comment (ijuma) [inline core/src/main/scala/kafka/server/BrokerServer.scala:393]: A couple of questions:
1. Where are we logging this for the zk case?
2. Can we expose a method in `KafkaConfig` to write the configs to a logger or something? Seems a bit cleaner than this approach.

Comment (cmccabe) [reply core/src/main/scala/kafka/server/BrokerServer.scala:393]: > 1. Where are we logging this for the zk case?

In the ZK case, it's logged from DynamicBrokerConfig.initialize (see above stack trace)

> 2. Can we expose a method in KafkaConfig to write the configs to a logger or something? Seems a bit cleaner than this approach.

Can we do this after 3.0? Due to the inheritance-based design here, there isn't a simple way to expose this without changing the base class. The comment does explain what the line does...

Comment (ijuma) [reply core/src/main/scala/kafka/server/BrokerServer.scala:393]: Can we file a JIRA?

Comment (ijuma) [reply core/src/main/scala/kafka/server/BrokerServer.scala:393]: I also learned that the ZK path logs dynamic configs too now. So, we can have a JIRA about logging dynamic configs for KRaft and mention there the potential clean-up.

**Review by ijuma — APPROVED** [review]
LGTM, thanks.

File: core/src/main/scala/kafka/server/BrokerServer.scala (hunk 1 above)
Comment (cmccabe) [reply core/src/main/scala/kafka/server/BrokerServer.scala:393]: Filed https://issues.apache.org/jira/browse/KAFKA-13105

**Pull request merged**